### PR TITLE
Rework built-in processing of JerryScript

### PR DESCRIPTION
**Pull request by zherczeg**
- All built-ins are native functions now
- Native handlers have a built-in id: ECMA_BUILTIN_ID_HANDLER
- Built-in routine identifiers start from 1
- Built-in routines have an own flag set
- Name property of routines is resolved dynamically
- Style fixes

### CODE REVIEW

File: jerry-core/ecma/builtin-objects/ecma-builtins.c [hunk 1]
```diff
@@ -376,11 +376,11 @@ ecma_instantiate_builtin (ecma_builtin_id_t obj_builtin_id) /**< built-in id */
 
   size_t property_count = ecma_builtin_get_property_count (obj_builtin_id);
 
-  if (property_count > 32)
+  if (property_count > 40)
```
Comment (rerobika) [inline jerry-core/ecma/builtin-objects/ecma-builtins.c:379]: Could you add a not about where does this 40 come from?

**Review by rerobika — COMMENTED** [review]
I like this patch, but could you please measure it on sunspider tests?

File: jerry-core/ecma/builtin-objects/ecma-builtins.c (hunk 1 above)
Comment (zherczeg) [reply jerry-core/ecma/builtin-objects/ecma-builtins.c]: I have added a macro for it

**Comment by ossy-szeged** [comment]
Benchmark | Heap (bytes) | Stack (bytes) | Perf (sec) |
----: | ----: | ----: | ----: | 
3d-cube.js | 10688 -> 10680 : +0.075% | 13620 -> 13680 : -0.441% | 1.157 -> 1.152 : +0.402% | 
3d-morph.js | 289016 -> 289008 : +0.003% | 1712 -> 1712 : 0.000% | 1.177 -> 1.201 : -1.953% | 
3d-raytrace.js | 119152 -> 119384 : -0.195% | 2548 -> 2548 : 0.000% | 1.319 -> 1.338 : -1.459% | 
access-binary-trees.js | 27120 -> 27120 : 0.000% | 3376 -> 3376 : 0.000% | 0.778 -> 0.767 : **+1.403%** | 
access-fannkuch.js | 1584 -> 1584 : 0.000% | 1784 -> 1784 : 0.000% | 2.581 -> 2.571 : +0.383% | 
access-nbody.js | 4088 -> 4088 : 0.000% | 1872 -> 1872 : 0.000% | 1.574 -> 1.585 : -0.704% | 
bitops-3bit-bits-in-byte.js | 808 -> 808 : 0.000% | 1728 -> 1728 : 0.000% | 0.693 -> 0.693 : +0.033% | 
bitops-bits-in-byte.js | 768 -> 768 : 0.000% | 1728 -> 1728 : 0.000% | 1.018 -> 1.015 : +0.278% | 
bitops-bitwise-and.js | 544 -> 544 : 0.000% | 1480 -> 1480 : 0.000% | 1.231 -> 1.234 : -0.203% | 
bitops-nsieve-bits.js | 60960 -> 60952 : +0.013% | 1728 -> 1728 : 0.000% | 1.476 -> 1.470 : +0.364% | 
controlflow-recursive.js | 1048 -> 1048 : 0.000% | 61592 -> 61592 : 0.000% | 0.517 -> 0.515 : +0.499% | 
crypto-aes.js | 27144 -> 27904 : -2.800% | 2384 -> 2288 : **+4.027%** | 0.884 -> 0.885 : -0.106% | 
crypto-md5.js | 70656 -> 70640 : +0.023% | 1796 -> 1796 : 0.000% | 0.729 -> 0.728 : +0.197% | 
crypto-sha1.js | 46000 -> 45984 : +0.035% | 1856 -> 1856 : 0.000% | 0.755 -> 0.752 : +0.359% | 
crypto.js | 50664 -> 50664 : 0.000% | 3164 -> 3164 : 0.000% | 8.747 -> 8.688 : +0.672% | 
date-format-tofte.js | 8696 -> 8688 : +0.092% | 2128 -> 2128 : 0.000% | 1.016 -> 1.010 : +0.579% | 
date-format-xparb.js | 14928 -> 14912 : +0.107% | 2696 -> 2688 : +0.297% | 0.817 -> 0.817 : +0.007% | 
deltablue.js | 455064 -> 453256 : +0.397% | 3548 -> 3540 : +0.225% | 5.427 -> 5.430 : -0.055% | 
math-cordic.js | 1824 -> 1824 : 0.000% | 1728 -> 1728 : 0.000% | 1.708 -> 1.703 : +0.243% | 
math-partial-sums.js | 1280 -> 1280 : 0.000% | 1704 -> 1704 : 0.000% | 1.053 -> 1.056 : -0.291% | 
math-spectral-norm.js | 3000 -> 2992 : +0.267% | 1728 -> 1728 : 0.000% | 0.738 -> 0.746 : -1.079% | 
raytrace.js | 21968 -> 21840 : +0.583% | 4748 -> 4952 : -4.297% | 2.793 -> 2.781 : +0.459% | 
richards.js | 8224 -> 8216 : +0.097% | 2132 -> 2148 : -0.750% | 0.308 -> 0.309 : -0.502% | 
string-base64.js | 89872 -> 89856 : +0.018% | 1776 -> 1776 : 0.000% | 1.411 -> 1.406 : +0.350% | 
string-fasta.js | 4168 -> 4056 : **+2.687%** | 1712 -> 1712 : 0.000% | 3.353 -> 3.394 : -1.199% | 
Geometric mean: | +0.059% | -0.030% | -0.050% | 

Binary (bytes) | master(3af3597f2e) | patch(366a051ec3) | Diff |
----: | ----: | ----: | ----: |
size | 239156 | 239156 | 0 bytes |
.rodata | 20703 | 20695 | -8 bytes |
.dynstr | 285 | 285 | 0 bytes |
.rel.plt | 240 | 240 | 0 bytes |
.interp | 25 | 25 | 0 bytes |
.dynsym | 576 | 576 | 0 bytes |
.gnu.hash | 252 | 252 | 0 bytes |
.text | 211024 | 211048 | +24 bytes |
.comment | 85 | 85 | 0 bytes |
.shstrtab | 226 | 226 | 0 bytes |
.data | 8 | 8 | 0 bytes |
.ARM.exidx | 8 | 8 | 0 bytes |
.rel.dyn | 16 | 16 | 0 bytes |
.init | 12 | 12 | 0 bytes |
.got | 136 | 136 | 0 bytes |
.plt | 384 | 384 | 0 bytes |
.note.ABI-tag | 32 | 32 | 0 bytes |
.gnu.version_r | 32 | 32 | 0 bytes |
.bss | 1575248 | 1575248 | 0 bytes |
.fini | 8 | 8 | 0 bytes |
.hash | 220 | 220 | 0 bytes |
.gnu.version | 72 | 72 | 0 bytes |
.fini_array | 4 | 4 | 0 bytes |
.init_array | 4 | 4 | 0 bytes |
.dynamic | 240 | 240 | 0 bytes |
.eh_frame | 4 | 4 | 0 bytes |
.ARM.attributes | 53 | 53 | 0 bytes |

**Review by rerobika — APPROVED** [review]
LGTM

**Review by dbatyai — APPROVED** [review]
LGTM

**Pull request merged**